### PR TITLE
GEOMESA-2905 Support CLI auto-ingest of Avro files with lenient schem…

### PR DESCRIPTION
**Pull request by elahrvivaz**
…a matching

Signed-off-by: Emilio Lahr-Vivaz <elahrvivaz@ccri.com>

### CODE REVIEW

File: geomesa-convert/geomesa-convert-avro/src/test/scala/org/locationtech/geomesa/convert/avro/AvroConverterTest.scala [hunk 1]
```diff
@@ -211,6 +211,34 @@ class AvroConverterTest extends Specification with AvroUtils with LazyLogging {
...
+      val features = Seq.tabulate(10) { i =>
+        ScalaSimpleFeature.create(sft, s"$i", s"name$i", i, s"2018-01-01T0$i:00:00.000Z", s"POINT(4$i 55)")
+      }
+
+      val out = new ByteArrayOutputStream()
+      WithClose(new AvroDataFileWriter(out, sft))(writer => features.foreach(writer.append))
+
+      val bytes = out.toByteArray
+
+      val updated = SimpleFeatureTypes.createType("test", "name:String,age:Int,dtg:Date,*geom:Point:srid=4326,tag:String")
```
Comment (jnh5y) [inline geomesa-convert/geomesa-convert-avro/src/test/scala/org/locationtech/geomesa/convert/avro/AvroConverterTest.scala:225]: Will this work with out of order columns?

Comment (elahrvivaz) [reply geomesa-convert/geomesa-convert-avro/src/test/scala/org/locationtech/geomesa/convert/avro/AvroConverterTest.scala:225]: yep! but we can't update schemas like that so i didn't add a test for it...

Comment (jnh5y) [reply geomesa-convert/geomesa-convert-avro/src/test/scala/org/locationtech/geomesa/convert/avro/AvroConverterTest.scala:225]: (Apologies, I think I'm going to end up asking my question twice:).)  Ok, but this Avro file would ingest fine into a datastore with the columns in a different order?

Comment (elahrvivaz) [reply geomesa-convert/geomesa-convert-avro/src/test/scala/org/locationtech/geomesa/convert/avro/AvroConverterTest.scala:225]: yep! tracks by name. you might end up with some errors if you have attributes with the same name but incompatible data types...

Comment (jnh5y) [reply geomesa-convert/geomesa-convert-avro/src/test/scala/org/locationtech/geomesa/convert/avro/AvroConverterTest.scala:225]: Yeah, I mean, if people are running that far off the rails...

**Pull request merged**